azext_iot/iothub/providers/base.py

```diff
@@ -29,3 +33,43 @@ def __init__(self, cmd, hub_name, rg, login=None, auth_type_dataplane=None):
 
     def get_sdk(self, sdk_type):
         return self.resolver.get_sdk(sdk_type)
+
+
+class IoTHubResourceProvider(object):
+    """Class to use for control plane operations."""
+    def __init__(
+        self,
+        cmd,
+        hub_name: str,
+        rg: Optional[str] = None,
+    ):
+        self.cmd = cmd
+        self.api_version = "2022-04-30-preview"
+        self.client = self.get_client()
+        self.discovery = IotHubDiscovery(cmd)
+        # Set discovery variables since we set client
+        self.discovery.track2 = True
+        self.discovery.client = self.client.iot_hub_resource
+        self.discovery.sub_id = get_subscription_id(self.cmd.cli_ctx)
+        # Need to get the direct resource
+        self.hub_resource = self.get_iot_hub_resource(hub_name, rg)
+
+    def get_client(self):
+        from azure.cli.core.commands.client_factory import get_mgmt_service_client
+        from azure.cli.core.profiles import ResourceType
+        client = get_mgmt_service_client(
+            self.cmd.cli_ctx,
+            ResourceType.MGMT_IOTHUB,
+            api_version=self.api_version
```

Comment:
Unclear if the optimal pattern is specifying the API version of the mgmt client here. If you don't specify one, you get the latest version correct?

Reply:
We resolved this in pair-programming /w @c-ryan-k